src/main/java/org/mule/extension/file/internal/command/LocalListCommand.java

```diff
@@ -68,14 +74,60 @@ public List<Result<InputStream, LocalFileAttributes>> list(FileConnectorConfig c
                                                              boolean recursive,
                                                              Predicate<LocalFileAttributes> matcher,
                                                              Long timeBetweenSizeCheck) {
+    return list(config, directoryPath, recursive, matcher, timeBetweenSizeCheck, null);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public List<Result<InputStream, LocalFileAttributes>> list(FileConnectorConfig config,
+                                                             String directoryPath,
+                                                             boolean recursive,
+                                                             Predicate<LocalFileAttributes> matcher,
+                                                             Long timeBetweenSizeCheck,
+                                                             SubsetList subsetList) {
     Path path = resolveExistingPath(directoryPath);
     if (!Files.isDirectory(path)) {
       throw cannotListFileException(path);
     }
 
     List<Result<InputStream, LocalFileAttributes>> accumulator = new LinkedList<>();
     doList(config, path.toFile(), accumulator, recursive, matcher, timeBetweenSizeCheck);
+    if (subsetList != null) {
+      return limitAndOrder(accumulator, (LocalSubsetList) subsetList);
+    }
+    return accumulator;
+  }
 
+  private List<Result<InputStream, LocalFileAttributes>> limitAndOrder(List<Result<InputStream, LocalFileAttributes>> accumulator,
+                                                                       LocalSubsetList subsetList) {
+    Integer offset = subsetList.getOffset();
+    Integer limit = subsetList.getLimit();
+    checkArgument(limit >= 0,
+                  String.format("Subset attribute '%s' must be greater than or equal to zero but '%d' was received", "limit",
+                                limit));
+    checkArgument(offset >= 0,
+                  String.format("Subset attribute '%s' must be greater than or equal to zero but '%d' was received", "offset",
+                                offset));
+    if (limit == 0) {
+      limit = accumulator.size();
+    }
+    if (offset == 0) {
+      offset = 1;
```

Comment:
what happen if offset is set to 1 externally? would return the same subset than setting it to 0?